modules/cudf/src/series.ts

```diff
@@ -1269,39 +1304,87 @@ function inferType(value: any[]): DataType {
+                                      |Column<T>|arrow.Vector<T>               //
+                                      |(string | null | undefined)[]           //
+                                      |(number | null | undefined)[]           //
+                                      |(bigint | null | undefined)[]           //
+                                      |(boolean | null | undefined)[]          //
+                                      |(Date | null | undefined)[]             //
+                                      |(string | null | undefined)[][]         //
+                                      |(number | null | undefined)[][]         //
+                                      |(bigint | null | undefined)[][]         //
+                                      |(boolean | null | undefined)[][]        //
+                                      |(Date | null | undefined)[][]): Column<T>;
+
+function asColumn<T extends DataType>(value: any) {
   if (value instanceof AbstractSeries) { return value._col; }
   if (Array.isArray(value)) {
     return fromArrow(arrow.Vector.from(
              {type: inferType(value), values: value as any, highWaterMark: Infinity})) as any;
   }
+
+  if (value instanceof Int8Array || value instanceof Int8Buffer) {
+    return new Column({type: new Int8, data: value, length: value.length});
+  } else if (value instanceof Int16Array || value instanceof Int16Buffer) {
+    return new Column({type: new Int16, data: value, length: value.length});
+  } else if (value instanceof Int32Array || value instanceof Int32Buffer) {
+    return new Column({type: new Int32, data: value, length: value.length});
+  } else if (value instanceof BigInt64Array || value instanceof Int64Buffer) {
+    return new Column({type: new Int64, data: value, length: value.length});
+  } else if (value instanceof Uint8Array || value instanceof Uint8Buffer) {
+    return new Column({type: new Uint8, data: value, length: value.length});
+  } else if (value instanceof Uint8ClampedArray || value instanceof Uint8ClampedBuffer) {
+    return new Column({type: new Uint8, data: value, length: value.length});
+  } else if (value instanceof Uint16Array || value instanceof Uint16Buffer) {
+    return new Column({type: new Uint16, data: value, length: value.length});
+  } else if (value instanceof Uint32Array || value instanceof Uint32Buffer) {
+    return new Column({type: new Uint32, data: value, length: value.length});
+  } else if (value instanceof BigUint64Array || value instanceof Uint64Buffer) {
+    return new Column({type: new Uint64, data: value, length: value.length});
+  } else if (value instanceof Float32Array || value instanceof Float32Buffer) {
+    return new Column({type: new Float32, data: value, length: value.length});
+  } else if (value instanceof Float64Array || value instanceof Float64Buffer) {
+    return new Column({type: new Float64, data: value, length: value.length});
+  }
```

Comment:
```
_|___|___|___|___|___|___|___|___|___|___|___|
___|___|___|___|___|___|___|___|___|___|___|_
_|___|___|___|___|___|___|___|___|___|___|___|
___|___|___|___|___|___|___|___|___|___|___|_
_|___|___|___|___|___|___|___|___|___|___|___|
___|___|___|___|___|___|___|___|___|___|___|_
_|___|___|___|___|___|___|___|___|___|___|___
```

Reply:
Honestly this is fine with me. If you prefer, you can make it a switch-case (like below). Neither is particularly pleasing, but like I mentioned earlier today, probably not worth agonizing over.

```suggestion
  switch (value.constructor) {
    case Int8Buffer:
    case Int8Array: return new Column({type: new Int8, data: value});
    case Int16Buffer:
    case Int16Array: return new Column({type: new Int16, data: value});
    case Int32Buffer:
    case Int32Array: return new Column({type: new Int32, data: value});
    case Int64Buffer:
    case BigInt64Array: return new Column({type: new Int64, data: value});
    case Uint8Buffer:
    case Uint8Array: return new Column({type: new Uint8, data: value});
    case Uint16Buffer:
    case Uint16Array: return new Column({type: new Uint16, data: value});
    case Uint32Buffer:
    case Uint32Array: return new Column({type: new Uint32, data: value});
    case Uint64Buffer:
    case BigUint64Array: return new Column({type: new Uint64, data: value});
    case Float32Buffer:
    case Float32Array: return new Column({type: new Float32, data: value});
    case Float64Buffer:
    case Float64Array: return new Column({type: new Float64, data: value});
    default: break;
  }
```